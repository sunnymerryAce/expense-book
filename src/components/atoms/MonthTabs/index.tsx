import React from "react";
import { makeStyles, useTheme } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Tabs from "@material-ui/core/Tabs";
import Tab from "@material-ui/core/Tab";

interface Props {
  months: Array<string>;
}

const tabParam = (index: number): Record<string, any> => {
  return {
    id: `scrollable-auto-tab-${index}`,
    "aria-controls": `scrollable-auto-tabpanel-${index}`,
  };
};

const MonthTabs: React.FC<Props> = (props) => {
  const theme = useTheme();
  const tabsWrapperClasses = makeStyles({
    root: {
      flexGrow: 1,
      width: "100%",
      backgroundColor: theme.palette.background.paper,
    },
  })();
  const [value, setValue] = React.useState(0);

  const handleChange = (
    event: React.ChangeEvent<{}>,
    newValue: number
  ): void => {
    console.log(newValue);
    setValue(newValue);
  };

  return (
    <div className={tabsWrapperClasses.root}>
      <AppBar position="static" color="default">
        <Tabs
          value={value}
          onChange={handleChange}
          indicatorColor="primary"
          textColor="primary"
          variant="scrollable"
          scrollButtons="auto"
        >
          {props.months.map((item, index) => {
            return <Tab key={index} label={item} {...tabParam(index)} />;
          })}
        </Tabs>
      </AppBar>
    </div>
  );
};

export default MonthTabs;
