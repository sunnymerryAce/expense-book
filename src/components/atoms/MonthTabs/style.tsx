import { makeStyles, createStyles, Theme } from "@material-ui/core/styles";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1,
      width: "100%",
      backgroundColor: theme.palette.background.paper,
    },
  })
);

export default useStyles;
