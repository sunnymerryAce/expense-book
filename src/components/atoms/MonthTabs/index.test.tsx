import React from "react";
import MonthTabs from "./";
import { shallow } from "enzyme";

describe("<MonthTabs />", () => {
  let container: any = null;
  const props = {
    months: [
      "202003",
      "20200401T000000",
      "20200501T000000",
      "20200601T000000",
      "Wed Jul 01 2020",
    ],
    currentIndex: 0,
    handleChange: jest.fn(),
  };

  beforeEach(() => {
    container = shallow(<MonthTabs {...props} />);
  });

  afterEach(() => {
    container.unmount();
    container = null;
  });

  test("should match the snapshot", () => {
    expect(container.html()).toMatchSnapshot();
  });

  test("event handler should be triggered", () => {
    container
      .find(".MonthTabs-Tabs")
      .simulate("change", { target: { value: "" } });
    expect(props.handleChange).toBeCalled();
  });
});
