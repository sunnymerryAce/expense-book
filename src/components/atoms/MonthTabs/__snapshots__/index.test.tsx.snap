// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MonthTabs /> should match the snapshot 1`] = `"<div class=\\"MuiBox-root MuiBox-root-3 makeStyles-root-1\\"><header class=\\"MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorDefault MuiPaper-elevation4\\"><div class=\\"MuiTabs-root MonthTabs-Tabs\\"><div style=\\"width:99px;height:99px;position:absolute;top:-9999px;overflow:scroll\\" class=\\"MuiTabs-scrollable\\"></div><div class=\\"MuiTabs-scroller MuiTabs-scrollable\\" style=\\"overflow:hidden\\"><div class=\\"MuiTabs-flexContainer\\" role=\\"tablist\\"><button class=\\"MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected\\" tabindex=\\"0\\" type=\\"button\\" role=\\"tab\\" aria-selected=\\"true\\" id=\\"scrollable-auto-tab-0\\"><span class=\\"MuiTab-wrapper\\">202003</span><span class=\\"PrivateTabIndicator-root-67 PrivateTabIndicator-colorPrimary-68 MuiTabs-indicator\\"></span></button><button class=\\"MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary\\" tabindex=\\"0\\" type=\\"button\\" role=\\"tab\\" aria-selected=\\"false\\" id=\\"scrollable-auto-tab-1\\"><span class=\\"MuiTab-wrapper\\">Wed Apr 01 2020</span></button><button class=\\"MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary\\" tabindex=\\"0\\" type=\\"button\\" role=\\"tab\\" aria-selected=\\"false\\" id=\\"scrollable-auto-tab-2\\"><span class=\\"MuiTab-wrapper\\">Fri May 01 2020</span></button><button class=\\"MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary\\" tabindex=\\"0\\" type=\\"button\\" role=\\"tab\\" aria-selected=\\"false\\" id=\\"scrollable-auto-tab-3\\"><span class=\\"MuiTab-wrapper\\">Mon Jun 01 2020</span></button><button class=\\"MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary\\" tabindex=\\"0\\" type=\\"button\\" role=\\"tab\\" aria-selected=\\"false\\" id=\\"scrollable-auto-tab-4\\"><span class=\\"MuiTab-wrapper\\">Wed Jul 01 2020</span></button></div></div></div></header></div>"`;
